### ISSUE TITLE
Squash cause *time.Time set to zero

### ISSUE THREAD

**Issue by wenerme** [mitchellh/mapstructure#228]
```golang
type A struct {
	AA
}
type AA struct {
	T *time.Time
}
func TestMapSquash(t *testing.T) {
	v := time.Now()
	in := &AA{
		T: &v,
	}
	out := &A{}
	d, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
		Squash: true,
		Result: out,
	})
	assert.NoError(t, err)
	assert.NoError(t, d.Decode(in))
	// these failed
	assert.True(t, v.Equal(*out.T))
	assert.False(t, out.T.IsZero())
}
```

**Comment by wenerme** [comment]
in `decodeMapFromStruct` will decode the time.Time to map, but there is no exported field, so and empty map is returned, this happened before Hook, no way to handle this.

**Comment by mitchellh** [comment]
This is fixed with #232 will release a patch release. Thank you VERY much for the test.

**Comment by anistark** [comment]
When's the release for this fix coming? @mitchellh

**Comment by mitchellh** [comment]
Just released, 1.4.1
